opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -93,26 +103,47 @@
 
         /**
          * A shorthand for addReference(References.CHILD_OF, parent).
+         *
+         * If parent==null, this is a noop.
          */
         SpanBuilder asChildOf(SpanContext parent);
 
         /**
          * A shorthand for addReference(References.CHILD_OF, parent.context()).
+         *
+         * If parent==null, this is a noop.
          */
-        SpanBuilder asChildOf(Span parent);
+        SpanBuilder asChildOf(BaseSpan parent);
```

Comment:
`SpanBuilder asChildOf(BaseSpan<?> parent)` should be good enough as the return type is independent from the actual generic type of `BaseSpan`.